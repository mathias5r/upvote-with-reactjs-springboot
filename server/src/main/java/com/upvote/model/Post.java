package com.upvote.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;

@Entity
public class Post {
	@Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    private long id;
	
	@Column(length = 80, nullable = false)
	private String author;
	private String date;
	@Column(length = 250, nullable = false)
	@Lob
	private String text;
	private long upvotes;

	public Post() {}
	
	public long getID() { return this.id; }
	public void setID(long id) { this.id = id; }
	
	public String getAuthor() { return this.author; }
	public void setAuthor(String author) { this.author = author; }
	
	public String getDate() { return this.date; }
	public void setDate(String date) { this.date = date; }
	
	public String getText() { return this.text; }
	public void setText(String text) { this.text = text; }
	
	public long getUpvotes() { return this.upvotes; }
	public void setUpvotes(long value) { this.upvotes = value;  }

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + (int) (id ^ (id >>> 32));
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Post other = (Post) obj;
		if (id != other.id)
			return false;
		return true;
	};
	
	
	
}
