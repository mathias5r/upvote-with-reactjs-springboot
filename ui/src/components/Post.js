import React, { Component } from 'react';
import upvote from '../resources/upvote.png';
import {postUpvote} from './API'
import Alert from 'react-s-alert';

class Post extends Component{

	constructor(props) {
		super(props);
	  	this.state = {
	    	post:props.post
	  	};
	  	this.upvotePost = this.upvotePost.bind(this);
  	}

	upvotePost(id){
		const params = new URLSearchParams();
	    params.append('id', id);
	    postUpvote(params).then(res =>{
	    	const post = res.data;
	 		this.setState({ post });
	    }).catch(error =>{
	        Alert.error('Erro ao tentar executar upvote!',{
	        	position: 'bottom-right'
	        });
  		});
	}

	render(){
		return (
			<div>
				<div className="row">
					<div className="col-md-auto">
					 	<p>{this.state.post.author}</p>
					</div>
					<div className="col-md-auto">
					 	<p>{this.state.post.date}</p>
					</div>
				</div>
				<div className="text-box rounded">
					<textarea maxlength="250" className="form-control" rows="5" id="comment" value={this.state.post.text}></textarea>
				</div>
				<div className="outer">
					<input onClick={() => this.upvotePost(this.state.post.id)} className="upvotes" type="image" src={upvote} alt="upvote"/>
					<p className="upvotesvalue">{this.state.post.upvotes}</p>
				</div>
			</div>
		);
	}
}
export default Post;