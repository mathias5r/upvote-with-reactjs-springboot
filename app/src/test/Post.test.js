import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Post from '../components/Post'
import ShallowRenderer from 'react-test-renderer/shallow';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
Enzyme.configure({ adapter: new Adapter() });

const post = { id: 100,
author: 'Mathias',
text: 'Test',
date: '88/88/2018',
upvotes: 0 };

describe('Testing Post.js', () =>{
	it('should render correctly...', () =>{
		const renderer = new ShallowRenderer();
		renderer.render(<Post post={post} />);
	  	const tree = renderer.getRenderOutput();
  		expect(tree).toMatchSnapshot();
	})
	it('should call upvotePost...',() => {
    	const spy = jest.spyOn(Post.prototype, 'upvotePost');
		const wrapper = shallow(<Post post={post} />);
		wrapper.find('input').simulate('click');
	  	expect(spy).toHaveBeenCalled();
	});
});


