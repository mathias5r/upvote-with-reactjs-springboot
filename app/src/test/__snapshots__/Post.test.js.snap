// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Post.js should render correctly... 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-auto"
    >
      <p>
        Mathias
      </p>
    </div>
    <div
      className="col-md-auto"
    >
      <p>
        88/88/2018
      </p>
    </div>
  </div>
  <div
    className="text-box rounded"
  >
    <textarea
      className="form-control"
      id="comment"
      maxlength="250"
      rows="5"
      value="Test"
    />
  </div>
  <div
    className="outer"
  >
    <input
      alt="upvote"
      className="upvotes"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      type="image"
    />
    <p
      className="upvotesvalue"
    >
      0
    </p>
  </div>
</div>
`;
